pkg/registry/core/service/storage/transaction.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+// transaction represents something that may need to be finalized on success or
+// failure of the larger transaction.
+type transaction interface {
+	// Commit tells the transaction to finalize any changes it may have
+	// pending.
+	Commit()
```

Comment:
So how would commit/revert report errors?

Reply:
They don't.  I have no idea what to do if your commit fails - the REST operation is already done.  So I make it your problem.

Reply:
so we expect to do 

t := transaction{
commit : func(){
// handle error here?
}
}